code/socialDistribution/requests.py

```diff
@@ -47,10 +50,24 @@ def post(url, params=None, body={}):
         "Accept": "application/json"
     }
 
-    response = requests.post(url, headers=headers, params=params, json=body)
+    
+    # Add Basic Auth Header specific to a node for Inbox api 
+    host = url_parser.get_host(url)
+    auth_credentials = node_manager.get_host_credentials(host)
+    if (sendBasicAuthHeader and auth_credentials):
+        authToken = base64.b64encode(auth_credentials).decode("ascii")
+        headers['Authorization'] = 'Basic %s' %  authToken
+    
+    try:
+        response = requests.post(url, headers=headers, params=params, json=body)
+
+        if response.status_code == 200:
+            return response.json() 
+        
+        return None
+    
+    except Exception as error:
+        print(error) 
```

Comment:
logging instead of print statement would be nice (see examples in api/views.py/inbox)

Reply:
Yes, thanks!